code-challenges.test.js

```diff
@@ -11,38 +13,135 @@
+//   > 17 |     it("should return an array of the Fibonacci sequence”, () => {
+//        |        ^
+//     18 |         const fibLength1 = 6
+//     19 |         const fibLength2 = 10
+//     20 |     expect(fibonacci(fibLength1)
+
+   
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
-// a) Create a test with expect statements for each of the variables provided.
+describe("oddNumbers", () => {
+    it("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+        const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+        expect(oddNumbers(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(oddNumbers(fullArr2)).toEqual([-823, 7, 23])
+})
+})
+
+
+// // a) Create a test with expect statements for each of the variables provided.
 
 const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+// // Expected output: [-9, 7, 9, 199]
 
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
+// // Expected output: [-823, 7, 23]
 
 
-// b) Create the function that makes the test pass.
+// // b) Create the function that makes the test pass.
+
+const oddNumbers = (nums) => {
+    for(let i=0; i<nums.length; i++){
+        if(nums[i] % 2 === 1) {
+            return []
+        }
+
+}
+}
+//Psuedocode:
+// 1. Declare a function called oddNumbers
+// 2. Create a parameter called nums
+// 3. Create a for loop with a modulo to get a remainder which would result in the odd numbers
+// 4. Create return statment
+
+// ----------------Notes question 2-------------------
+// unsure about the numbers in the strings, is the question looking for strictly data type numbers or string is included? unsure of what method to use as far as including strings or being able to sort from smallest to largest, I was thinking in the return line just write out all the odd numbers in that order or a .split & .join method but unsure on what to add for the reassignment. Can one parameter be used for 2 variables? 
+
+//error message:
+// ✕ takes in an array and returns a new array of only odd numbers sorted from least to greatest
+//   addition
+//     ✕  takes in an array and returns an array of the accumulating sum.
+
+//   ● oddNumbers › takes in an array and returns a new array of only odd numbers sorted from least to greatest
+
+//     ReferenceError: array is not defined
+
+//       90 |
+//       91 | const oddNumbers = (nums) => {
+//     > 92 |     for(let i=0; i<array.length; i++){
+//          |                    ^
+//       93 |         if(array[i] % 2 === 1) {
+//       94 |             return []
+//       95 |         }
+
+//-------------- so after the message I switched it to 'nums' and got the same response
```

Comment:
I like where you're going with the logic!  You're getting the error because you used nums as the parameter, but then use array in the function.